### PR TITLE
ovn: add testcase for startup pod reconciliation

### PR DESCRIPTION
**Pull request by dcbw**
@danwinship @shettyg @girishmg @pecameron @squeed

### CODE REVIEW

File: go-controller/pkg/ovn/pods_test.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+				Expect(err).NotTo(HaveOccurred())
+
+				_, err = config.InitConfig(ctx, fexec, nil)
+				Expect(err).NotTo(HaveOccurred())
+
+				fakeClient := fake.NewSimpleClientset(&v1.PodList{
+					Items: []v1.Pod{*newPod(namespace, podName, nodeName)},
+				})
+
+				err = util.SetExec(fexec)
```
Comment (danwinship) [inline go-controller/pkg/ovn/pods_test.go]: You do `InitConfig` + `SetExec` twice here (both before and after creating `fakeClient`). Cut and paste error or something? (Likewise in the other tests.)

File: go-controller/pkg/ovn/pods_test.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+	}
+}
+
+func newPod(namespace, name, node string) *v1.Pod {
+	return &v1.Pod{
+		Status: v1.PodStatus{
+			Phase: v1.PodRunning,
+		},
+		ObjectMeta: newObjectMeta(namespace, name),
+		Spec: v1.PodSpec{
```
Comment (danwinship) [inline go-controller/pkg/ovn/pods_test.go]: (nitpick: `ObjectMeta`, `Spec`, `Status`. Likewise in `newNamespace`)

**Comment by dcbw** [comment]
@danwinship PTAL thanks!

**Comment by danwinship** [comment]
lgtm, except that `newNamespace` still has the weird field ordering

**Pull request merged**